sway-core/src/semantic_analysis/ast_node/expression/typed_expression.rs

```diff
@@ -516,6 +519,16 @@ impl TypedExpression {
                 },
                 span,
             ),
+            Expression::BuiltinGenerateB256Seed { span } => ok(
```

Comment:
What about writing the equivalent of "GenerateB256Seed" as a function in the standard library? That way we don't have to special case it in the compiler.

Reply:
The compiler passes through the intrinsic `__generate_b256_seed` (name is WIP) all the way to codegen where the actual generation happen. I'm not sure we can do that with pure library code and without a new intrinsic, right?